Source/JavaScriptCore/llint/WebAssembly.asm

```diff
@@ -605,16 +605,29 @@ op(js_to_wasm_wrapper_entry, macro ()
     const AccumulatorTag = invalidGPR # Only used for JSVALUE32_64
 
 if ARM64 or ARM64E or ARMv7
-    const CP = ws3 # Callee prologue (jump target) pointer
+    const MPC = t8
+    const CalleeExecutableAddress = ws3 # Callee prologue (jump target) pointer
     const Accumulator = ws0
     const Scratch = ws1
     const Scratch2 = ws2
+
+    const MPCSpill = -0x28
 else
+    const MPC = metadataTable
     const Accumulator = csr2
     const Scratch = ws1
+
+    # Any callee saves clobbered here would also have to be saved by the JS->WASM thunk,
+    # since the exception unwinder cannot tell if an interpreted callee had a replacementCallee installed
+    # when the stack frame was created.
+    # Instead of making every thunk save more callee-saves, we should just restore these extra
+    # clobbered callee-saves here before calling JS.
```

Comment:
Do you mean that we would replace the interpreted callee directly, instead of via m_replacementCallee? Or that we would mark weather the frame had a replacementCallee when it was created?

Reply:
The former, the upside would be twofold. 1) We wouldn't need to match the spills between the two JSToWasm callees. 2) We could actually free lower tier Wasm code that's been replaced as the GC'd Callee on the stack would keep the JIT code alive while it is still reachable.

Reply:
Nice, sure

Reply:
https://bugs.webkit.org/show_bug.cgi?id=274768